File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int answer = nums[0];
        int curSum = nums[0];
        
        for (int i = 1; i < nums.length; i++) {
            int num = nums[i];
            curSum = Math.max(curSum + num, num);
            answer = Math.max(answer, curSum);
        }
        
        return answer;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let answer = nums[0];
    let curSum = 0;
    
    for (const num of nums) {
        if (curSum < 0) {
            curSum = 0;
        }
        curSum += num;
        answer = Math.max(answer, curSum);
    }
    
    return answer;
};